File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/KeyValueStore.java
package com.garpr.android.preferences;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.Map;

public interface KeyValueStore {

    void clear();

    boolean contains(@NonNull final String key);

    @Nullable
    Map<String, ?> getAll();

    boolean getBoolean(@NonNull final String key, final boolean fallbackValue);

    float getFloat(@NonNull final String key, final float fallbackValue);

    int getInteger(@NonNull final String key, final int fallbackValue);

    long getLong(@NonNull final String key, final long fallbackValue);

    String getString(@NonNull final String key, @Nullable final String fallbackValue);

    void remove(@NonNull final String key);

    void setBoolean(@NonNull final String key, final boolean value);

    void setFloat(@NonNull final String key, final float value);

    void setInteger(@NonNull final String key, final int value);

    void setLong(@NonNull final String key, final long value);

    void setString(@NonNull final String key, @NonNull final String value);

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/KeyValueStore.kt
package com.garpr.android.preferences

interface KeyValueStore {

    val all: Map<String, *>?

    fun batchEdit(): BatchEditor

    fun clear()

    operator fun contains(key: String): Boolean

    fun getBoolean(key: String, fallbackValue: Boolean): Boolean

    fun getFloat(key: String, fallbackValue: Float): Float

    fun getInteger(key: String, fallbackValue: Int): Int

    fun getLong(key: String, fallbackValue: Long): Long

    fun getString(key: String, fallbackValue: String?): String?

    fun remove(key: String)

    fun setBoolean(key: String, value: Boolean)

    fun setFloat(key: String, value: Float)

    fun setInteger(key: String, value: Int)

    fun setLong(key: String, value: Long)

    fun setString(key: String, value: String)

    interface BatchEditor {
        fun apply()

        fun clear(): BatchEditor

        fun putBoolean(key: String, value: Boolean): BatchEditor

        fun putInteger(key: String, value: Int): BatchEditor

        fun putString(key: String, value: String): BatchEditor
    }

}
